cala-ledger/src/balance/repo.rs

```diff
@@ -12,14 +13,99 @@ use std::collections::{HashMap, HashSet};
 
 #[derive(Debug, Clone)]
 pub(super) struct BalanceRepo {
-    _pool: PgPool,
+    pool: PgPool,
 }
 
 impl BalanceRepo {
     pub fn new(pool: &PgPool) -> Self {
-        Self {
-            _pool: pool.clone(),
+        Self { pool: pool.clone() }
+    }
+
+    pub async fn find(
+        &self,
+        journal_id: JournalId,
+        account_id: AccountId,
+        currency: Currency,
+    ) -> Result<AccountBalance, BalanceError> {
+        let row = sqlx::query!(
+            r#"
+            SELECT h.values, a.normal_balance_type AS "normal_balance_type!: DebitOrCredit"
+            FROM cala_balance_history h
+            JOIN cala_current_balances c
+            ON h.data_source_id = c.data_source_id
+            AND h.journal_id = c.journal_id
+            AND h.account_id = c.account_id
+            AND h.currency = c.currency
+            AND h.version = c.latest_version
+            JOIN cala_accounts a
+            ON c.data_source_id = a.data_source_id
+            AND c.account_id = a.id
+            WHERE c.data_source_id = '00000000-0000-0000-0000-000000000000'
```

Comment:
why this has been hardcoded ?

Reply:
Querying for synced entities is not on the critical path for our target use case. Hard coding this now to simplify. There are a few ways to deal with querying non-local entities and I'm not sure yet which approach is the right one.... will decide that later when relevant and restrict queries to local entities for now.